lib/utils/ensure-single-browser-tab-only.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import BootWarning from '../components/boot-warning';
+
+const HEARTBEAT_DELAY = 1000;
+const clientId = uuidv4();
+const emptyLock = [null, -Infinity];
+const foundElectron = window.process && window.process.type;
+
+if (!foundElectron) {
+  if ('lock-acquired' !== grabSessionLock()) {
+    ReactDOM.render(<BootWarning />, document.getElementById('root'));
+    throw new Error('Simplenote can only be opened in one tab');
+  }
+  let keepGoing = true;
+  loop(() => {
+    if (!keepGoing) {
+      return false;
+    }
+    switch (grabSessionLock()) {
+      case 'lock-acquired':
+        return true; // keep updating the lock and look for other sessions which may have taken it
+
+      default:
+        window.alert(
+          "We've detected another session running Simplenote. This shouldn't be. Expect problems. A refresh will help."
```

Comment:
cc: @SylvesterWilmott for copy text and the general UX question of "what do we do when we've been running fine and suddenly realize that another fighting copy of the app has loaded?"

Reply:
Just to clarify, am I right in assuming I would see this message in the first instance of Simplenote if I opened a new instance in another browser window?

Reply:
This "should never occur" because if our lock is working then the second window will abort its boot and display the message "Can't open two tabs"

There are cases where this could occur, such as when processing note updates take exceptionally long, when an extension interferes in some unpredictable way, when someone clears `localStorage` in their browser, or when something else unexpected breaks our locking mechanism.

Reply:
"We've detected another session running Simplenote, expect problems to occur. Please refresh the page."

Reply:
"We've detected another session running Simplenote, this may cause problems while editing notes. Please refresh the page."